text/0000-global-packages.md

```diff
@@ -0,0 +1,195 @@
+
+This has the benefit of providing a clear way to upgrade the pinned Node if a user desires: Switch their default and re-install the package. While that looks a bit like other package managers—which require you to re-install packages for every different version of Node—the key difference is that with Volta, this path is entirely _optional_ and not required unless you want to change the version of Node associated with a package.
+
+### Handling Multiple Installs
+
+If the user calls `npm i -g <package_1> <package_2>`, that would normally all run as a single command. However, to maintain Volta's Sandbox-per-package behavior, we will need to split that out and run `npm i -g <package_1>` and `npm i -g <package_2>` separately, allowing us to install them in separate locations and avoid combining them into a single block.
+
+### Note on `pnpm`
+
+`pnpm` can also be redirected with a `prefix` configuration setting, however we also need to do some messing with the PATH when calling `pnpm i -g`. The reason is that `pnpm` searches the `PATH` to find the highest-precedence "appropriate" location to install the binary symlinks. Unfortunately, that currently means in the Volta image directory for the specific Node version, which is definitely _not_ where we want them. We will likely need to create a directory and add it to the PATH for `pnpm` global installs, to make sure that the symlinks are created in the correct location.
+
+## 2. Shared Global Package Directory
+
+In order to satisfy constraints 4 & 5, allowing global packages to `require` each other, we should create a single shared "globals" directory that we add to `NODE_PATH` when executing a global binary (and _only_ for global binaries, as normal Node commands wouldn't have access the globals anyway). That directory can have symlinks to all of the individual package directories, so that a `require` from any one of the packages will be able to access any of the others. Per the [Node documentation](https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders), the `NODE_PATH` will only be searched if the regular filesystem process fails, so this should result in the correct precedence when resolving a global library.
+
+With the shared package directory available, we can then remove the restriction that we currently enforce that prevents installing global libraries.
+
+Finally, we should document the location of this shared package directory, so that if users want to include it in their own configurations (e.g. for one-off scripts), they can easily locate the correct directory.
```

Comment:
If we expect people to use it in scripts it's probably a good idea to have a `volta` command (like `volta info globals-dir`) that can spit out the proper path, so that can be the "public API" for relying on the directory, making it easier for users and at the same time easier for us to potentially change the location in the future.

Reply:
That’s a good call, it would definitely be good to expose that in the same way that `volta which` unwraps the shims.

Reply:
Included a call-out that a new command would be helpful for this.